src/js/components/notifications/index.js

```diff
@@ -0,0 +1,82 @@
+import { BasicBlock } from "../basic-block";
+import { getCustomCheckbox } from "../../utils/element-factory";
+
+import { subscribe } from "../messaging/subscribe";
+
+function setInAppNotificationState($customCheckbox, $checkbox) {
+  navigator.serviceWorker.ready
+    .then((serviceWorkerRegistration) => {
+      serviceWorkerRegistration.pushManager
+        .getSubscription()
+        .then((subscription) => {
+          // the user is subscribed so, unsubscribe
+          if (subscription) {
+            subscription
+              .unsubscribe()
+              .then((result) => {
+                if (result) {
+                  localStorage.setItem("hasSubscription", false);
+                  $customCheckbox.removeClass("w--redirected-checked");
+                  $checkbox.checked = false;
+                }
+              })
+              .catch((error) => {
+                console.error(
+                  `Error while unsubscribing from subscription: ${error.toString()}`
+                );
+              });
+          } else {
+            subscribe($customCheckbox, $checkbox);
+          }
+        })
+        .catch((error) => {
+          console.error(
+            `Error while getting subscription: ${error.toString()}`
+          );
+        });
+    })
+    .catch((error) => {
+      console.error(`Error in serviceWorker ready: ${error.toString()}`);
+    });
+}
+
+export const NotificationsSettings = () => {
+  const $inAppNotificationsContainer = getCustomCheckbox({
+    identifier: "notify-inapp",
+    name: "notify-inapp",
+    text: "Receive in-app notifications",
+  });
+  const $inAppNotificationsCutomCheckbox = $inAppNotificationsContainer.find(
+    ".w-checkbox-input"
+  );
+  const $inAppNotificationsCheckbox = $inAppNotificationsContainer.find(
+    "input[type='checkbox']"
+  );
+  const $inAppNotificationsNoSupportMsg = new BasicBlock({
+    title: "In-app notifications is not supported on your device",
+    subtitle: "",
+  });
+  // is Notifications and ServiceWorker supported?
+  if (!("Notification" in window) && !("serviceWorker" in navigator)) {
```

Comment:
shouldn't this be || ?
Should we also test for push support?

Reply:
Funny you should ask that. I originally had it as an `or`. Now I am thinking perhaps instead of testing for not supported we should test for support so, something like:

```
// pseudo code

if notifications && serviceWorker && pushMessage
  then ....
else
  your device ....
```

Thoughts? I know you generally want to return early.

Reply:
!A && !B is equivalent to A || B so which way around we do it doesn't matter too much.

But currently this says "if neither notifications, nor service workers are supported, say we don't support"
What we mean to say is "If either of notifications or services workers are not supported, say we don't support. I'll create a ticket

Reply:
https://trello.com/c/2ltpUB6T/173-strengthen-requirements-for-indicator-notifications-support